src/Controller/Api/Notification/NotificationBaseApi.php

```diff
@@ -110,6 +111,13 @@ protected function serializeNotification(Notification $dto)
                 $ban = $dto->getSubject();
                 $toReturn['subject'] = $this->magazineFactory->createBanDto($ban);
                 break;
+            case 'report_created_notification':
+            case 'report_rejected_notification':
+            case 'report_approved_notification':
+                /** @var ReportCreatedNotification $n */
+                $n = $dto;
+                $toReturn['reportId'] = $n->report->getId();
+                break;
```

Comment:
The notification structure has the actual subject of the notification detailed - using the report id instead breaks this convention, and would also make it impossible for regular users to have any information about the report contained in this notification displayed by the app using the API, since retrieving reports is a moderator and admin only API endpoint.
This solution would not be sufficient if we wanted to display the reason a user's post was deleted in the notification using the API, for example

Reply:
What I would recommend would be to follow the convention of the other notifications and embed a subset of the report in the notification, at least the `reason`, `magazine`, `type`, and `subjectId` of the report
`subjectId` would be the post/comment/entry that was reported and removed

Reply:
I'll check that

Reply:
Ok, here are my thoughts:

- `ReportCreatedNotification`: only admins and mods can get this, so imo its fine to only send the id of the report, but of course a dto of the report could be added for this.
- `ReportRejectedNotification`: a user should be able to see their own reports, but if not maybe add the subject of the report here, as there is no reason supplied for rejecting a report.
- `ReportApprovedNotification`: since this is going to the author of the reported content, the subject that was reported should be enough. 

Would you be fine with that?

The reason I do not just want to add a report dto to all of them is simple: you should never be informed who is reporting your content for which reason, unless the report is accepted, but even then the reason the reporting user supplied is not necessarily the one the mod removed your content for. Since we do not have a "moderators_reason" field (or something similar) it would not make much sense to just relay that information to the reported user